core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -261,9 +263,24 @@ protected SqlCall as(SqlNode e, String alias, String... fieldNames) {
 
   /** Returns whether a list of expressions projects all fields, in order,
    * from the input, with the same names. */
-  public static boolean isStar(List<RexNode> exps, RelDataType inputRowType,
+  public static boolean isStar(List<RexNode> exps, RelNode input,
       RelDataType projectRowType) {
     assert exps.size() == projectRowType.getFieldCount();
+    // Duplicate column names
+    // For example, EMP JOIN DEPT, both of which have columns named DEPTNO
+    if (input instanceof Join || input instanceof MultiJoin) {
```

Comment:
Encapsulate this part of the code into an independent function, maybe it named `isJoinDuplicateName`.

Call function by `!isStar() && !isJoinDuplicateName()`.

Reply:
@JiajunBernoulli Thank you for your advice. I've considered adding a method before, but it might still cause an error when the isStar method is called from somewhere else. If the method name of 'isStar' is changed to something like 'projectAllField', `!projectAllField() && ! isJoinDuplicateName()` would that be better?
If you insist that the method name not be changed, I will also take your advice.

Reply:
Oh, combined with the first comment, I get it, because it's a public static method, so the method name won't change. Then I'll take your advice.